scalatest/src/main/scala/org/scalatest/EitherValues.scala

```diff
@@ -98,6 +107,32 @@ trait EitherValues {
    */
   implicit def convertRightProjectionToValuable[L, R](rightProj: Either.RightProjection[L, R])(implicit pos: source.Position): RightValuable[L, R] = new RightValuable(rightProj, pos)
 
+  class EitherValuable[L, R](either: Either[L, R], pos: source.Position) {
+    /**
+      * Returns the <code>Left</code> value contained in the wrapped <code>LeftProjection</code>, if defined as a <code>Left</code>, else throws <code>TestFailedException</code> with
+      * a detail message indicating the <code>Either</code> was defined as a <code>Right</code>, not a <code>Left</code>.
+      */
+    def leftValue: L = {
+      try {
+        either.left.get
+      }
+      catch {
+        case cause: NoSuchElementException =>
+          throw new TestFailedException((_: StackDepthException) => Some(Resources.eitherLeftValueNotDefined), Some(cause), pos)
+      }
+    }
+
+    /**
+      * Returns the <code>Right</code> value contained in the wrapped <code>Either</code>, if defined as a <code>Right</code>, else throws <code>TestFailedException</code> with
+      * a detail message indicating the <code>Either</code> was defined as a <code>Left</code>, not a <code>Right</code>.
+      */
+    def rightValue: R = either match {
+      case Right(x) => x
```

Comment:
and the same here, just `either.getOrElse(throw new TestFailedException(...))` should work.

Reply:
`.getOrElse` is nicer than the pattern matching, fixed! :)